at_lookup/lib/src/at_lookup_impl.dart

```diff
@@ -267,19 +267,31 @@ class AtLookupImpl implements AtLookUp {
     }
     // If response starts with error:, throw AtLookupException.
     if (_isError(verbResult)) {
-      verbResult = verbResult.replaceAll('error:', '');
-      // Setting the errorCode and errorDescription to default values.
-      var errorCode = 'AT0014';
-      var errorDescription = 'Unknown server error';
-      if (verbResult.contains('-')) {
-        if (verbResult.split('-')[0].isNotEmpty) {
-          errorCode = verbResult.split('-')[0];
-        }
-        if (verbResult.split('-')[1].isNotEmpty) {
-          errorDescription = verbResult.split('-')[1];
+      verbResult = verbResult.replaceFirst('error:', '');
+      try {
+        var decodedResponse = jsonDecode(verbResult);
+        throw AtExceptionUtils.get(
+            decodedResponse['errorCode'], decodedResponse['errorDescription'])
+          ..stack(Triple(
+              Intent.remoteVerbExecution,
+              ExceptionScenario.remoteVerbExecutionFailed,
+              decodedResponse['errorDescription']));
+        // If JSON decoding fails, fall back to older version
+        // Parse the error response in throw AtLookupException.
+      } on FormatException {
```

Comment:
I don't know how expensive it is to try to decode and then throw a FormatException. Could perhaps reduce the likelihood of that happening by checking first of all if the verbResult starts with a '{'? (and only trying to jsonDecode if it does indeed look like json)

Reply:
Added IF condition to verify if result startsWith '{' and endsWith '}' to check if response is JSON encoded string.